### PR TITLE
Frontend lazy handling support 

### PR DESCRIPTION
**Pull request by jcorbin**
Building on the work of #1239:
- allow operations to handle frames lazily (for ongoing established calls)
- allow channel to handle frames lazily (for new incoming calls)

r @kriskowal @Raynos

### CODE REVIEW

**Comment by jcorbin** [comment]
Backed out the node/hyperbahn changes, tchannel only, ready for review please.

File: node/service-name-handler.js [hunk 1]
```diff
@@ -42,6 +42,49 @@ function TChannelServiceNameHandler(options) {
...
+            error: res.err
+        }));
+        // TODO: protocol error instead?
+        self._sendLazyErrorFrame(conn, reqFrame, 'BadRequest', 'failed to read serviceName');
+        return false;
+    }
+
+    var serviceName = res.value;
+    if (!serviceName) {
+        self.channel.logger.warn('missing service name in lazy frame', conn.extendLogInfo({}));
```
Comment (Raynos) [inline node/service-name-handler.js]: lets try and pull out callerName & arg1.

Can we implement `reqFrame.extendLogInfo()` ?

**Comment by Raynos** [comment]
lgtm.

Left comment about `LazyReq.extendLogInfo` as an alternative to `InRequest.extendLogInfo`

File: node/service-name-handler.js [hunk 2]
```diff
@@ -42,6 +42,49 @@ function TChannelServiceNameHandler(options) {
...
+    if (chan && chan.handler.handleLazily) {
+        return chan.handler.handleLazily(conn, reqFrame);
+    } else {
+        return false;
+    }
+};
+
+TChannelServiceNameHandler.prototype._sendLazyErrorFrame =
+function _sendLazyErrorFrame(conn, reqFrame, codeString, message) {
+    var fakeR = {
```
Comment (kriskowal) [inline node/service-name-handler.js]: R is for response. Please at least spell out Res.

**Comment by kriskowal** [comment]
:ok:

**Pull request merged**